sys/net/routing/rpl/Makefile

```diff
@@ -1,8 +1,16 @@
-MODE = $(shell grep "RPL_DEFAULT_MOP RPL_NON_STORING_MODE" ../../include/rpl/rpl_config.h)
-ifneq (,$(MODE))
-	DIRS += rpl_nonstoring
+MODE = $(shell grep "\#define $(RPL_MOP)" ../../include/rpl/rpl_config.h | sed 's/\#define \s*//g')
```

Comment:
to be more precise on this line, `MODE` is filled with a lot of garbage != `RPL_MOP_* 0x**` if `$RPL_MOP` is not set.  
In the next lines below it is tested if we have only a pair of words in `MODE`, consisting of a MOP define and its according number, e.g. `RPL_MOP_NO_DOWNWARD_ROUTES 0x00`.
If we have only a pair, the according number (word 2) is used to set the `RPL_DEFAULT_MOP` define